Library/Homebrew/api.rb

```diff
@@ -15,6 +12,10 @@ module API
 
     extend Cachable
 
+    autoload :Analytics, "api/analytics"
```

Comment:
Is this provided by ActiveSupport? If so, it may be worth weighing up whether this provides more speed increases than removing ActiveSupport entirely does. 

CC @dduugg who has been working in this area for thoughts.

Reply:
Most of the remaining ActiveSupport stuff should be monkey-patches, which I don't think can benefit from `autoload`.

Amazing stuff here though! Looking very promising.

Reply:
@dduugg Sorry, I more meant: is `autoload` a native Ruby feature or an ActiveSupport one? It seems to be a Ruby one, I thought it might have been an ActiveSupport one.

Reply:
Ah, yes, it's part of Ruby core: https://ruby-doc.org/core-2.6.10/Module.html#method-i-autoload

Reply:
There's another shorter variation included in active support from what I remember but I'm using the Ruby core version.